ion-schema/src/constraint.rs

```diff
@@ -281,13 +274,6 @@ impl Constraint {
         }))
     }
 
-    /// Creates a [Constraint::ValidValues] using the [Range] specified inside it
-    pub fn valid_values_with_range(value: Range) -> Constraint {
```

Comment:
There are now two types of ranges for valid_values, so it would need to be changed to `valid_values_with_number_range` and `valid_values_with_timestamp_range`.

... or instead, you should be able to do this:
```rust
let my_range: NumberRange = something();
let cons1 = valid_values(vec![my_range.into()]);

let my_ts_range: TimestampRange = something_else();
let cons2 = valid_values(vec![my_ts_range.into()]);
```

Basically, it's extra surface area in our APIs, and I don't think it adds a lot of value.

Reply:
Yes, I think keep the `valid_values()` and using it as you mentioned sounds good to me.